smsclient_core/models/sms_gateway.py

```diff
@@ -0,0 +1,220 @@
+# coding: utf-8
+# Copyright (C) 2004-2009 Tiny SPRL (<http://tiny.be>).
+# Copyright (C) 2011 SYLEAM (<http://syleam.fr/>)
+# Copyright (C) 2013 Julius Network Solutions SARL <contact@julius.fr>
+# Copyright (C) 2015 Valentin Chemiere <valentin.chemiere@akretion.com>
+# Copyright (C) 2015 Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+import logging
+_logger = logging.getLogger(__name__)
+
+PRIORITY_LIST = [
+    ('0', '0'),
+    ('1', '1'),
+    ('2', '2'),
+    ('3', '3')
+]
+
+CLASSES_LIST = [
+    ('0', 'Flash'),
+    ('1', 'Phone display'),
+    ('2', 'SIM'),
+    ('3', 'Toolkit')
+]
+
+
+class SMSClient(models.Model):
+    _name = 'sms.gateway'
+    _description = 'SMS Client'
+
+    name = fields.Char(string='Gateway Name', required=True)
+    from_provider = fields.Char(string="From")
+    method = fields.Selection(string='API Method', selection=[])
+    url = fields.Char(
+        string='Gateway URL', help='Base url for message')
+    state = fields.Selection(selection=[
+        ('new', 'Not Verified'),
+        ('waiting', 'Waiting for Verification'),
+        ('confirm', 'Verified'),
+        ], string='Gateway Status', index=True, readonly=True, default='new')
+    user_ids = fields.Many2many(
+        comodel_name='res.users',
+        string='Users Allowed')
+    code = fields.Char('Verification Code')
+    body = fields.Text(
+        string='Message',
+        help="The message text that will be send along with the"
+             " email which is send through this server.")
+    validity = fields.Integer(
+        default=10,
+        help="The maximum time - in minute(s) - before the message "
+             "is dropped.")
+    classes = fields.Selection(
+        selection=CLASSES_LIST, string='Class',
+        default='1',
+        help='The SMS class: flash(0),phone display(1),SIM(2),toolkit(3)')
```

Comment:
IMO repeating the options in the help doesn't add much value, and disallows the changing of items without redeclaring the field